homeassistant/components/hunterdouglas_powerview/select.py

```diff
@@ -0,0 +1,125 @@
+"""Support for hunterdouglass_powerview settings."""
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+import logging
+from typing import Any, Final
+
+from aiopvapi.resources.shade import BaseShade, factory as PvShade
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_BATTERY_KIND,
+    DOMAIN,
+    POWER_SUPPLY_TYPE_MAP,
+    POWER_SUPPLY_TYPE_REVERSE_MAP,
+    ROOM_ID_IN_SHADE,
+    ROOM_NAME_UNICODE,
+    SHADE_BATTERY_LEVEL,
+)
+from .coordinator import PowerviewShadeUpdateCoordinator
+from .entity import ShadeEntity
+from .model import PowerviewDeviceInfo, PowerviewEntryData
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PowerviewSelectDescriptionMixin:
+    """Mixin to describe a select entity."""
+
+    current_fn: Callable[[BaseShade], Any]
+    options_fn: Callable[[BaseShade], list[Any]]
+    select_fn: Callable[[BaseShade, str], Coroutine[Any, Any, bool]]
+
+
+@dataclass
+class PowerviewSelectDescription(
+    SelectEntityDescription, PowerviewSelectDescriptionMixin
+):
+    """A class that describes select entities."""
+
+    entity_category: EntityCategory = EntityCategory.CONFIG
+
+
+DROPDOWNS: Final = [
+    PowerviewSelectDescription(
+        key="powersource",
+        name="Power Source",
+        icon="mdi:power-plug-outline",
+        current_fn=lambda shade: POWER_SUPPLY_TYPE_MAP.get(
+            shade.raw_data.get(ATTR_BATTERY_KIND)
+        ),
+        options_fn=lambda shade: list(POWER_SUPPLY_TYPE_MAP.values()),
```

Comment:
Since it's static you should be able to define options in the description instead

That was recently added see you may need to rebase

Reply:
arent they only static because we have a single select dropdown ? ie - would you still have them in the description dataclass if there were two select options implemented ?

Reply:
Yes.  Examples:

https://github.com/home-assistant/core/blob/bbe63bca4733b31b6f6cf29270cdd62765309fdf/homeassistant/components/xiaomi_miio/select.py#L152
https://github.com/home-assistant/core/blob/dev/homeassistant/components/senseme/select.py#L52
https://github.com/home-assistant/core/blob/dev/homeassistant/components/lifx/select.py#L19

Reply:
no worries - just commiting those changes now